pxr/usd/lib/usd/stage.cpp

```diff
@@ -1722,6 +1722,36 @@ UsdStage::GetPrimAtPath(const SdfPath &path) const
     return UsdPrim(primData, proxyPrimPath);
 }
 
+UsdObject
+UsdStage::GetObjectAtPath(const SdfPath &path) const
+{
+    // Maintain consistent behavior with GetPrimAtPath
+    if (!path.IsAbsolutePath()) {
+        return UsdObject();
+    }
+
+    const bool isPrimPath = path.IsPrimPath();
+    const bool isPropPath = !isPrimPath && path.IsPropertyPath();
+    if (!isPrimPath && !isPropPath) {
+        return UsdObject();
+    }
+
+    // A valid prim must be found to return either a prim or prop
+    if (isPrimPath) {
+        if (auto prim = GetPrimAtPath(path)) {
+            return prim;
+        }
+    } else if (isPropPath) {
+        if (auto prim = GetPrimAtPath(path.GetPrimPath())) {
+            if (auto prop = prim.GetProperty(path.GetNameToken())) {
```

Comment:
Similar here; we need to check the validity of the prim, but we should be able to just return prim.GetProperty(...) and not need to check the validity of the property object before returning it.